src/MultiFormats/Cryptography/KeccakManaged.cs

```diff
@@ -0,0 +1,355 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+// The SHA3 doesn't create .Net Standard package.
+// This is a copy of https://bitbucket.org/jdluzen/sha3/raw/d1fd55dc225d18a7fb61515b62d3c8f164d2e788/SHA3Managed/SHA3Managed.cs
+
+using System;
+
+namespace MultiFormats.Cryptography
+{
+    internal partial class KeccakManaged : Keccak
+    {
+        public KeccakManaged(int hashBitLength)
+            : base(hashBitLength) { }
+
+        protected override void HashCore(byte[] array, int ibStart, int cbSize)
+        {
+            base.HashCore(array, ibStart, cbSize);
+            if (cbSize == 0)
+                return;
+            var sizeInBytes = SizeInBytes;
+            if (buffer == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)